components/scream/src/share/grid/remap/horizontal_remap_utility.hpp

```diff
@@ -0,0 +1,564 @@
+#ifndef EAMXX_HORIZONTAL_REMAP_UTILITY_HPP
+#define EAMXX_HORIZONTAL_REMAP_UTILITY_HPP
+
+#include "share/grid/abstract_grid.hpp"
+#include "share/io/scorpio_input.hpp"
+#include "share/io/scream_scorpio_interface.hpp"
+#include "share/scream_types.hpp"
+#include "ekat/ekat_workspace.hpp"
+
+#include <numeric>
+
+namespace scream {
+
+
+/* --------------------------------------------------------------------------------------------- */
+/* A simple structure to store remap information for a single target column
+ * --------------------------------------------------------------------------------------------- */
+struct RemapSegment {
+
+using gid_type = AbstractGrid::gid_type;
+using Pack = ekat::Pack<Real,SCREAM_PACK_SIZE>;
+
+using KT = KokkosTypes<DefaultDevice>;
+
+template <typename S>
+using view_1d = typename KT::template view_1d<S>;
+
+template <typename S>
+using view_2d = typename KT::template view_2d<S>;
+
+template <typename S>
+using view_3d = typename KT::template view_3d<S>;
+
+template <typename S>
+using view_1d_host = typename KT::template view_1d<S>::HostMirror;
+
+template <typename S>
+using view_2d_host = typename KT::template view_2d<S>::HostMirror;
+
+public:
+  virtual ~RemapSegment() = default;
```

Comment:
Do you intend for this to be a base class with subclasses inheriting from it? If not, you don't need a `virtual` destructor.